### DISCUSSION TITLE
Daten via MQTT setzen

### DISCUSSION THREAD

**Discussion by heckmic** [syssi/esphome-smg-ii#40] [General]
Ich würde gerne den "Maximum mains charging current" via mqtt setzen.
Wenn ich das in iobroker machen, dann springt der beim nächsten auslesen immer automatisch auf den Wert von vorher. Es wurde also nichts gesetzt.

**Comment by syssi** [comment]
Was genau tust du um den neuen Wert zu schreiben? An welches Topic schickst du welchen Wert?

**Reply by syssi** [reply]
Ich sehe gerade, dass

```
Maximum mains charging current                        0.1A    Uint  333 1 R/W
```

bisher nur als `sensor` implementiert ist und nicht als `number`. Deshalb ist dieser Wert bisher nur lesbar und nicht schreibbar. Ich bereite eine Aenderung vor, so dass man den Wert in Zukunft auch schreiben darf.

**Reply by syssi** [reply]
Ich habe die Beispiel-Konfiguration (YAML) erweitert. Es gibt nun `number`-Entitäten, über welche du die Einstellung des Inverters verändern kannst. Die Sensor-Entitäten sind read-only und deshalb nicht per MQTT veränderbar.

**Reply by Quickmick666** [reply]
Hi, ich habe meine 3 SMGs mit deiner Software verbunden und lese die Daten über MQTT mit FHEM aus. Leider bekomme ich es nicht hin Daten zu senden.
Folgendes habe ich versucht:
smg_usb_sued2_4022d876def8:smg-usb-sued2/select/smg-usb-sued2_output_priority/PV-Battery-Utility (SBU)

smg-usb-sued2/select/smg-usb-sued2_output_priority/state PV-Battery-Utility (SBU)

smg-usb-sued2/select/smg-usb-sued2_output_priority/2

leider ohne Erfolg.

**Reply by syssi** [reply]
Hier findest du die exakten Commands: https://github.com/syssi/esphome-smg-ii/issues/69#issuecomment-2330891197

Das Topic muss auf `/command` enden. Der Wert muss aus dem linken Teil der Optionmap stammen also ein String sein (keine Ganzzahl).
